src/component/term/TermMetadataCreateForm.tsx

```diff
@@ -47,20 +46,13 @@ export class TermMetadataCreateForm extends React.Component<TermMetadataCreateFo
         }
     }
 
-    public componentDidUpdate(prevProps: TermMetadataCreateFormProps, prevState: TermMetadataCreateFormState): void {
-        if (this.props.language && (prevProps.language !== this.props.language)) {
-            this.onPrefLabelChange( this.props.termData.label[this.props.language] || '' );
-        }
-    }
-
     private onLabelChange = (e: React.ChangeEvent<HTMLInputElement>) => {
         this.onPrefLabelChange(e.currentTarget.value);
     };
 
     private onPrefLabelChange = (label : string) => {
         const change = Object.assign({}, this.props.termData.label);
         change[this.props.language] = label;
-        this.props.onChange({label: change});
```

Comment:
I'm sorry for keeping this blocked, but:
- Why do you create a `change` object, which is not used anywhere besides the assignment
- And then use the `checkLabelUniqueness` method, which IMHO should be a validation-only function, for the side-effect of updating the parent state via the onChange method?